lbrynet/lbrynet_daemon/LBRYDaemonServer.py

```diff
@@ -179,12 +361,35 @@ def render_GET(self, request):
     #     call.addErrback(lambda err: log.info("Error: " + str(err)))
     #     call.cancel()
 
+class LBRYFileUpload(resource.Resource):
+    """
+    Accepts a file sent via the file upload widget in the web UI, saves
+    it into a temporary dir, and responds with a JSON string containing
+    the path of the newly created file.
+    """
+
+    def __init__(self, api):
+        self._api = api
+
+    def render_POST(self, request):
+        origfilename = request.args['file_filename'][0]
+        uploaded_file = request.args['file'][0]  # Temp file created by request
+
+        # Move to a new temporary dir and restore the original file name
+        newdirpath = tempfile.mkdtemp()
+        newpath = os.path.join(newdirpath, origfilename)
+        shutil.move(uploaded_file.name, newpath)
+        self._api.uploaded_temp_files.append(newpath)
+
+        return json.dumps(newpath)
+
 
 class LBRYDaemonServer(object):
     def _setup_server(self, wallet):
         self.root = LBRYindex(os.path.join(os.path.join(data_dir, "lbry-ui"), "active"))
         self._api = LBRYDaemon(self.root, wallet_type=wallet)
         self.root.putChild("view", HostedLBRYFile(self._api))
+        self.root.putChild("upload", LBRYFileUpload(self._api))
```

Comment:
The thing is that it's not an endpoint in the main JSON-RPC API, it's a POST resource for a form in the web UI. We wouldn't want to mix that functionality in with the JSON-RPC code, ditto for "view."